llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -12250,10 +12260,11 @@ void SelectionDAGBuilder::lowerWorkItem(SwitchWorkListItem W, Value *Cond,
         }
         break;
       }
+      case CC_And:
       case CC_Range: {
         const Value *RHS, *LHS, *MHS;
         ISD::CondCode CC;
-        if (I->Low == I->High) {
+        if (I->Low == I->High || I->Kind == CC_And) {
```

Comment:
This change is odd since `CC_And` already implies that `Low == High`. Maybe it would be better to add an assert in the `else` case that `Kind != CC_And`?

Reply:
Thanks, added assert and removed the check. That was a left-over from an earlier version.